pkg/v1/mutate/mutate.go

```diff
@@ -92,6 +93,49 @@ func AppendManifests(base v1.ImageIndex, adds ...IndexAddendum) v1.ImageIndex {
 	}
 }
 
+// RemoveManifests removes any descriptors that match the match.Matcher.
+func RemoveManifests(base v1.ImageIndex, matcher match.Matcher) (v1.ImageIndex, error) {
+	ii, err := base.IndexManifest()
+	if err != nil {
+		return base, err
+	}
+
+	// create a list without any existing ones
+	manifests := []v1.Descriptor{}
+	for _, manifest := range ii.Manifests {
+		if !matcher(manifest) {
+			manifests = append(manifests, manifest)
+		}
+	}
+	return &index{
+		base:     base,
+		replaces: manifests,
+	}, nil
+}
+
+// ReplaceManifests removes any descriptors that match the match.Matcher and replaces with new ones.
+// It is the equivalent of calling RemoveManifests().AppendManifests(), but a bit more efficient and
+// convenient.
+func ReplaceManifests(base v1.ImageIndex, matcher match.Matcher, adds ...IndexAddendum) (v1.ImageIndex, error) {
```

Comment:
I'm not a huge fan of these semantics -- I'd prefer we drop this for now and just have `layout` call RemoveManifests + AppendManifests. If you're worried about efficiency, we could do some typechecking to avoid some of this work, but I don't imagine it's _that_ expensive to recompute the manifest.

Reply:
I am not sure I understand. Are you suggesting that we just get rid of `ReplaceManifests()` and leave it to the equivalent (which is referenced in the comment)?

Reply:
> Are you suggesting that we just get rid of ReplaceManifests() and leave it to the equivalent (which is referenced in the comment)?

Yeah exactly. From my original comment:

> For your replace implementation, I would really expect:
> 
> `Layout([A, B, C]).ReplaceImage(D, Matcher(B)) == Layout([A, D, C])` and also
> 
> `Layout([A, A]).ReplaceImage(B, Matcher(A)) == Layout([B, B])`,
> 
> but these are currently just equivalent to `Append(Remove(Matcher(...), X)`.
> 
> If just Remove then Append is your desired behavior, how would you feel about adding Remove to mutate, then doing the remove && append in your client code? I'd be okay with a "replace" for mutate if it did in-place and not just remove + append.

I don't think we agree on what `ReplaceManifests` should do, and since what you want to do here is achievable without it, I'd rather punt on `ReplaceManifests` and just make the `layout.ReplaceDescriptor` implementation slightly more verbose.

Reply:
So basically we are just getting rid of `ReplaceManifests()`. OK.

Reply:
OK, I did that and removed the one usage of it.